crates/sui-indexer-alt-jsonrpc/src/api/coin.rs

```diff
@@ -308,6 +334,22 @@ async fn filter_coins(
     })
 }
 
+async fn coin_metadata_object_id(
+    ctx: &Context,
+    coin_type: String,
+) -> Result<Option<ObjectID>, RpcError<Error>> {
+    let coin_type_param = StructTag::from_str(&coin_type)
+        .map_err(|e| invalid_params(Error::BadType(coin_type.clone(), e)))?;
+
+    let full_coin_type = sui_types::coin::CoinMetadata::type_(coin_type_param);
+
+    let object_id = load_singleton_object_id(ctx.pg_reader(), full_coin_type)
+        .await
+        .context("Failed to load singleton object id")?;
+
+    Ok(object_id)
```

Comment:
Let's inline this last call (the variable name is in any case slightly misleading because it's an `Option<ObjectID>`, not an `ObjectID`).

```suggestion
    Ok(load_singleton_object_id(ctx.pg_reader(), full_coin_type)
        .await
        .context("Failed to load singleton object id")?)
```